x-pack/legacy/plugins/siem/server/lib/detection_engine/types.ts

```diff
@@ -74,6 +70,9 @@ export type RuleTypeParams = Omit<
   'name' | 'enabled' | 'interval' | 'tags' | 'actions' | 'throttle'
 >;
 
+export type RuleTypeParamsWithName = RuleTypeParams & {
```

Comment:
Why the special casing of `name`, here? Is it possible to use it throughout our SIEM alerts, or do we expressly not want it in other alerts?